# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/moz.build
# Commit: 13f8ed81dca8
# Full Hash: 13f8ed81dca894a7b5c274d2c33bc4c984702bff
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Introduce DecoderAgent r=alwu
#   
#   DecoderAgent is a class operating MediaDataDecoder. It combines multiple
#   MediaDataDecoder APIs into one to serve WebCodecs' needs.
#   
# ==============================================================================

diff -r d62002584cf3 -r 13f8ed81dca8 dom/media/webcodecs/moz.build
--- a/dom/media/webcodecs/moz.build	Thu Jun 29 22:33:11 2023 +0000
+++ b/dom/media/webcodecs/moz.build	Thu Jun 29 22:33:12 2023 +0000
@@ -6,6 +6,16 @@
 
 MOCHITEST_MANIFESTS += ["test/mochitest.ini"]
 
+# For mozilla/layers/ImageBridgeChild.h
+LOCAL_INCLUDES += [
+    "!/ipc/ipdl/_ipdlheaders",
+    "/ipc/chromium/src/",
+]
+
+EXPORTS.mozilla += [
+    "DecoderAgent.h",
+]
+
 EXPORTS.mozilla.dom += [
     "EncodedVideoChunk.h",
     "VideoColorSpace.h",
@@ -15,6 +25,7 @@
 ]
 
 UNIFIED_SOURCES += [
+    "DecoderAgent.cpp",
     "EncodedVideoChunk.cpp",
     "VideoColorSpace.cpp",
     "VideoDecoder.cpp",
